scripts/integration/Dockerfile

```diff
@@ -0,0 +1,5 @@
+FROM rust:1.57-slim-bullseye
+
+RUN apt-get update \
+  && apt-get install -y g++ cmake pkg-config libclang1-9 llvm-9 libsasl2-dev libssl-dev \
+  && rm -rf /var/lib/apt/lists/*
```

Comment:
The other solution would be to pass the version as a build argument 🤔

Reply:
We use a similar base for the soak tests which I bumped recently in https://github.com/vectordotdev/vector/pull/10472/files. I believe rustup will still do the right thing regardless when `cargo` is run? That is, if the version doesn't match the toolchain file, install it.

Reply:
A build argument could be good though. It would be nice to be able to tie it back to `rust-toolchain.toml` to avoid needing to update a bunch of places every time we upgrade Rust.

Reply:
😆 `echo rust-toolchain.toml`...

Reply:
`cat rust-toolchain.toml | grep channel | cut -d '"' -f 2` 😅